core/application/impl/app_state_manager_impl.cpp

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "application/impl/app_state_manager_impl.hpp"
+
+#include <csignal>
+#include <functional>
+
+namespace kagome {
+
+  std::weak_ptr<AppStateManager> AppStateManagerImpl::wp_to_myself;
+
+  AppStateManagerImpl::AppStateManagerImpl()
+      : logger_(common::createLogger("Application")) {
+    struct sigaction act {};
+    memset(&act, 0, sizeof(act));
+    act.sa_handler = shuttingDownSignalsHandler;  // NOLINT
+    sigset_t set;
```

Comment:
You might use boost's [signal_set](https://www.boost.org/doc/libs/1_47_0/doc/html/boost_asio/reference/signal_set.html)  instead

Reply:
No, it is not appropriate here. Boost handle signals asinchronously in his own context. But we need to control running boost context running here.